scripts/oe-env-prepare.ps1

```diff
@@ -0,0 +1,446 @@
+        $parameters['FilePath'] = 'msiexec.exe'
+        $parameters['ArgumentList'] = @("/i", $InstallerPath) + $ArgumentList
+    }
+    Write-Output "Installing $InstallerPath"
+    $p = Start-Process @parameters
+    if($p.ExitCode -ne 0) {
+        Throw "Failed to install: $InstallerPath"
+    }
+    Add-ToSystemPath -Path $EnvironmentPath
+    Write-Output "Successfully installed: $InstallerPath"
+}
+
+
+function Install-ZipTool {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$ZipPath,
+        [Parameter(Mandatory=$true)]
+        [string]$InstallDirectory,
+        [Parameter(Mandatory=$false)]
+        [string[]]$EnvironmentPath
+    )
+    if(Test-Path $InstallDirectory) {
+        Write-Output "$ZipPath is already installed."
+        Add-ToSystemPath -Path $EnvironmentPath
+        return
+    }
+    New-Item -ItemType "Directory" -Path $InstallDirectory
+    7z.exe x $ZipPath -o"$InstallDirectory" -y
+    if($LASTEXITCODE) {
+        Throw "ERROR: Failed to extract $ZipPath to $InstallDirectory"
+    }
+    Add-ToSystemPath $EnvironmentPath
+}
+
+
+function Install-Git {
+    $installDir = Join-Path $env:ProgramFiles "Git"
+    Install-Tool -InstallerPath $PACKAGES["git"]["local_file"] `
+                 -InstallDirectory $installDir `
+                 -ArgumentList @("/SILENT") `
+                 -EnvironmentPath @("$installDir\cmd", "$installDir\bin", "$installDir\mingw64\bin")
+
+}
+
+
+function Install-7Zip {
+    $installDir = Join-Path $env:ProgramFiles "7-Zip"
+    Install-Tool -InstallerPath $PACKAGES["7z"]["local_file"] `
+                 -InstallDirectory $installDir `
+                 -ArgumentList @("/quiet", "/passive") `
+                 -EnvironmentPath @($installDir)
+}
+
+
+function Install-SGX {
+    $installDir = Join-Path $PACKAGES_DIRECTORY "sgx_base"
+    Install-ZipTool -ZipPath $PACKAGES["sgx_drivers"]["local_file"] `
+                    -InstallDirectory $installDir
+    pnputil /add-driver "$installDir\sgx_base.inf"
+    if ($LASTEXITCODE -ne 0) {
+        throw "Failed to install SGX Drivers"
+    }
+}
+
+
+function Install-PSW {
+    $installDir = Join-Path $PACKAGES_DIRECTORY "intel_psw_install"
+    Install-ZipTool -ZipPath $PACKAGES["psw"]["local_file"] `
+                    -InstallDirectory $installDir
+
+    $pswInstaller = Join-Path $installDir "Intel SGX PSW for Windows v2.3.100.49777\PSW_EXE_RS2_and_before\Intel(R)_SGX_Windows_x64_PSW_2.3.100.49777.exe"
+    $p = Start-Process -Wait -NoNewWindow -PassThru -FilePath $pswInstaller `
+                       -ArgumentList @("--extract-folder", "$installDir", "--x")
+    if($p.ExitCode -ne 0) {
+        Throw "Failed to extract the Intel SGX PSW bundle: $pswInstaller"
+    }
+    $p = Start-Process -Wait -NoNewWindow -PassThru -FilePath "$installDir\install.exe" `
+                       -ArgumentList @("install", "--eula=accept", "--output=$installDir\intel_install.log", "--components=all")
```

Comment:
Please get the unattended parameters from our Ansible scripts:
https://github.com/microsoft/openenclave/blob/a982b46cf440def8fb66e94f2622a4f81e2b350b/scripts/ansible/roles/windows/az-dcap-client/tasks/environment-setup.yml#L64

We want to maintain exactly what's in Ansible.